shell/common/vsync_waiter.cc

```diff
@@ -46,48 +46,87 @@ void VsyncWaiter::AsyncWaitForVsync(Callback callback) {
       return;
     }
     callback_ = std::move(callback);
+    if (secondary_callback_) {
+      // Return directly as `AwaitVSync` is already called by
```

Comment:
This is unfortunate.  I wonder if there is some way we can enforce it gets called at least once either way.